pkg/migrator/migrator_test.go

```diff
@@ -87,15 +87,17 @@ func TestMigrator(t *testing.T) {
 	// Publishing to the blockchain takes time.
 	<-time.After(10 * time.Second)
 
-	checkMigrationTrackerState(t, test.ctx, test.db)
-	checkGatewayEnvelopesLastID(t, test.ctx, test.db)
-	checkGatewayEnvelopesMigratedAmount(t, test.ctx, test.db)
-	checkGatewayEnvelopesAreUnique(t, test.ctx, test.db)
+	require.Eventually(t, func() bool {
+		return checkMigrationTrackerState(t, test.ctx, test.db) &&
+			checkGatewayEnvelopesLastID(t, test.ctx, test.db) &&
+			checkGatewayEnvelopesMigratedAmount(t, test.ctx, test.db) &&
+			checkGatewayEnvelopesAreUnique(t, test.ctx, test.db)
+	}, 10*time.Second, 100*time.Millisecond)
```

Comment:
this can wait for 10-20s. Should we remove the wait?

Publishing to blockchain takes 250ms. Why does it take more than 10s?